dgl_ptm/dgl_ptm/model/initialize_model.py

```diff
@@ -315,21 +361,75 @@ def _initialize_agents_tec(self):
         for i in range(len(self.technology_levels)):
             technology_mask = agentsTecLevel == i
             agentsGamma[technology_mask] = self.gamma_vals[i]
-            agentsCost[technology_mask] = self.cost_vals[i]   
+            agentsCost[technology_mask] = self.cost_vals[i]
         return agentsTecLevel, agentsGamma, agentsCost
 
     def step(self):
         try:
             self.step_count +=1
-            ptm_step(self.model_graph,self.model_data,self.step_count,self.steering_parameters)
-        except:
-            #TODO add model dump here. Also check against previous save to avoid overwriting
+            print(f'performing step {self.step_count} of {self.step_target}')
+            ptm_step(self.model_graph, self.model_data, self.step_count, self.steering_parameters)
 
+            # store each step
```

Comment:
Storing the model at each step can have detrimental effects on the speed, as well as potentially being very storage intensive. It might be better to add a specification of the frequency (every x steps) with which a save state is created.

Reply:
it is fixed now. It only saves every x step.